service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -488,7 +488,7 @@ private String prepareCodeElement(Observation observation) {
             return codeableConceptCdMapper.mapCodeableConceptToCd(observation.getCode());
         }
 
-        return StringUtils.EMPTY;
+        throw new EhrMapperException("%s must contain a code element.".formatted(observation.getId()));
```

Comment:
Does `getId()` return something like `Observation/abc` ? If so then that looks good to me.

Reply:
Checked, and it does.